integration/metadata_test.rb

```diff
@@ -0,0 +1,152 @@
+    assert_equal :text, table_meta.clustering_columns[0].type.kind
+    assert_equal :asc, table_meta.clustering_order.first
+
+    assert_equal 4, table_meta.columns.size
+    table_meta.each_column do |column|
+      assert ['user_id', 'first', 'last', 'age'].any? { |name| name == column.name }
+      assert [:bigint, :text, :int].any? { |type| type == column.type.kind }
+    end
+  end
+
+  # Test for column ordering in table metadata
+  #
+  # test_column_ordering_is_deterministic tests that the metadata relating to the columns are retrieved in the proper
+  # order. This proper order is: partition key, clustering columns, and then all other columns alphanumerically.
+  #
+  # @since 3.0.0
+  # @jira_ticket RUBY-180
+  # @expected_result column ordering should be correct in table metadata.
+  #
+  # @test_category metadata
+  #
+  def test_column_ordering_is_deterministic
+    assert @cluster.keyspace('simplex').has_table?('users')
+    table_meta = @cluster.keyspace('simplex').table('users')
+    table_cql = Regexp.new(/CREATE TABLE simplex\.users \(
+  user_id bigint,
+  last text,
+  age int,
+  first text,
+  PRIMARY KEY \(user_id, last\)
+\)/)
+
+    assert_equal 0, table_meta.to_cql =~ table_cql
+
+    col_names = ['user_id', 'last', 'age', 'first']
+    table_meta.each_column do |column|
+      assert_equal col_names[0], column.name
+      col_names.delete_at(0)
+    end
+  end
+
+  # Test for retrieving crc_check_balance property
+  #
+  # test_table_metadata_contains_crc_check_balance tests that the 'crc_check_balance' property of table metadata is able
+  # to be retrieved.
+  #
+  # @since 3.0.0
+  # @jira_ticket RUBY-179
+  # @expected_result crc_check_balance property should be retrieved from the table metadata
+  #
+  # @test_category metadata
+  #
+  def test_table_metadata_contains_crc_check_balance
+    skip("The crc_check_balance property on a table was introduced in Cassandra 3.0") if CCM.cassandra_version < '3.0.0'
+
+    assert @cluster.keyspace('simplex').has_table?('users')
+    table_meta = @cluster.keyspace('simplex').table('users')
+    refute_nil table_meta.options.crc_check_chance
+    assert_equal 1.0, table_meta.options.crc_check_chance
+  end
+
+  # Test for retrieving extensions property
+  #
+  # test_table_metadata_contains_extensions tests that the 'extensions' property of table metadata is able to be
+  # retrieved.
+  #
+  # @since 3.0.0
+  # @jira_ticket RUBY-170
+  # @expected_result extensions property should be retrieved from the table metadata
+  #
+  # @test_category metadata
+  #
+  def test_table_metadata_contains_extensions
+    skip("The extensions property on a table was introduced in Cassandra 3.0") if CCM.cassandra_version < '3.0.0'
+
+    assert @cluster.keyspace('simplex').has_table?('users')
+    table_meta = @cluster.keyspace('simplex').table('users')
+    refute_nil table_meta.options.extensions
+    assert_empty table_meta.options.extensions
```

Comment:
If assert_empty doesn't accept nil, you can get rid of the refute_nil above, can't you? Or does assert_empty allow for nil?